src/Client.cpp

```diff
@@ -10,67 +10,45 @@
 #include "libursa/ZHelpers.h"
 #include "spdlog/spdlog.h"
 
-bool UrsaClient::wait_sec() {
-    // make small sleeps so the thread could
-    // terminate quickly if it's requested to do so
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
-    wait_time += 10;
-    return wait_time >= 1000;
-}
+void UrsaClient::check_task_status(const std::string &conn_id) {
+    s_send<std::string_view>(&status_socket, "status;");
+
+    std::string res_str;
+    constexpr int max_retries = 30;
+    for (int i = 0; i < max_retries && res_str.empty(); i++) {
+        try {
+            res_str = s_recv<std::string>(&status_socket);
```

Comment:
Is it somehow guaranteed that recv will wait exactly a second before returning?

Also, since we had a discussion about exceptions today, maybe we should hide this try/catch and use `s_try_recv` or something. Just an idea though

Reply:
Nvm the first part, I just saw timeout below

Reply:
:+1:  s_try_recv / s_try_send added